packages/lesswrong/components/form-components/SocialPreviewUpload.tsx

```diff
@@ -100,103 +141,224 @@ mdi.use(markdownItSup)
+        image: null,
+      };
+    }
+    const data = document.contents?.dataWithDiscardedSuggestions ?? contents.data;
+    const html = contents.type === "markdown" ? mdi.render(data, { docId: randomId() }) : data;
+    const parser = new DOMParser();
+    const htmlDoc = parser.parseFromString(html, "text/html");
+    const img = htmlDoc.querySelector("img");
+    const plaintextDescription = toPlainText(html, contents.type);
     return {
-      description: `<Description preview not supported for this editor type (${originalContents.type}), switch to HTML, Markdown, or ${ckEditorName} to see the description preview>`,
-      fallbackImageUrl: null,
+      description: plaintextDescription,
+      image: img?.getAttribute("src"),
     };
-  }
+  };
 
-  const data = dataWithDiscardedSuggestions ?? originalContents.data
-  const html = contentsType === 'markdown' ? mdi.render(data, {docId: randomId()}) : data
-  if (!html) return {description: null, fallbackImageUrl: null}
-
-  const parser = new DOMParser();
-  const htmlDoc = parser.parseFromString(html, "text/html");
-  // get first img tag
-  const img = htmlDoc.querySelector("img");
-
-  const truncatedHtml = truncate(sanitize(html), PLAINTEXT_HTML_TRUNCATION_LENGTH)
-  const plaintextDescription = htmlToText(truncatedHtml, {
-    wordwrap: false,
-    selectors: [
-      { selector: "img", format: "skip" },
-      { selector: "a", options: { ignoreHref: true } },
-      { selector: "p", options: { leadingLineBreaks: 1 } },
-      { selector: "h1", options: { trailingLineBreaks: 1 } },
-      { selector: "h2", options: { trailingLineBreaks: 1 } },
-      { selector: "h3", options: { trailingLineBreaks: 1 } },
-    ]
-  }).substring(0, PLAINTEXT_DESCRIPTION_LENGTH);
+  const originalContentProcessed = originalContents
+    ? processContents(originalContents)
+    : { description: null, image: null };
+  const highlightPlaintextDesc = customHighlight ? toPlainText(customHighlight.data, customHighlight.type) : null;
 
-  const previewDesc = getPostDescription({...document, contents: {plaintextDescription}})
-  
-  return {description: previewDesc, fallbackImageUrl: img?.getAttribute("src") || null}
-}
+  const previewDesc =
+    socialText !== undefined
+      ? socialText
+      : getPostDescription({
+          ...document,
+          contents: { plaintextDescription: originalContentProcessed.description },
+          customHighlight: { plaintextDescription: highlightPlaintextDesc },
+        });
+
+  return {
+    description: previewDesc,
+    fallbackImageUrl: originalContentProcessed.image ?? null,
+  };
+};
 
-const SocialPreviewUpload = ({name, document, updateCurrentValues, clearField, label, croppingAspectRatio, classes}: {
-  name: string,
-  document: PostsEditWithLocalData,
-  updateCurrentValues: UpdateCurrentValues,
-  clearField: Function,
-  label: string,
-  croppingAspectRatio?: number,
-  classes: ClassesType
+const SocialPreviewTextEdit = ({
```

Comment:
I haven't set any hard limit on the max length people can write. I expect common sense + the nudge of the field being kind of small will do the job, and there isn't really a problem with writing a really long paragraph here (some places like slack actually do show a lot of preview text)